src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -230,6 +231,33 @@ public IEnumerable<NuGetProject> GetNuGetProjects()
             return projects;
         }
 
+        public bool IsAllProjectsNominated()
+        {
+#if VS14
+            // for VS14, always return true since nominations don't apply there.
+            return true;
+#else
+            var netCoreProjects = GetNuGetProjects().OfType<NetCorePackageReferenceProject>().ToList();
+
+            // count for nominated projects so far
+            var nominatedProjects = 0;
```

Comment:
you don't have to count all of them, just find at least one without restore info.